src/history/hs.c

```diff
@@ -481,17 +481,22 @@ __hs_insert_record_with_btree(WT_SESSION_IMPL *session, WT_CURSOR *cursor, WT_BT
     WT_ASSERT(session, type == WT_UPDATE_STANDARD || type == WT_UPDATE_MODIFY);
 
     /*
-     * If the time points are out of order (which can happen if the application performs updates
-     * with out-of-order timestamps), so this value can never be seen, don't bother inserting it.
+     * If the application performs updates with mixed mode transactions, this value can never be
```

Comment:
I think mixed mode timestamps is a better phrasing, and perhaps an explanation of what "mixed mode" timestamps are would be helpful in this comment, i.e. having a zero timestamp which effectively makes the rest of the chain obsolete. 

Mixed mode transactions makes me think we're running at different isolation levels or something.